plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -114,7 +114,7 @@ func Run(s *options.SchedulerServer) error {
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
 			Namespace: "kube-system",
-			Name:      "kube-scheduler",
+			Name:      "kube-" + s.SchedulerName,
```

Comment:
+1 to leaving the default lock object the same (by whatever mechanism)

not sure about using the scheduler name as the lock object's name... there are valid annotation names that are not valid endpoint names, and I wouldn't want to accidentally stomp an existing endpoint. I still think letting them pick the endpoint to lock on (both namespace and name) is valuable.